### PR TITLE
Add watch task to gulp

### LINKED ISSUE #5: Add gulp-watch task
Add a task in `gulpfile.js` to use `gulp-watch` to watch directories and run relevant tasks.

### PR DESCRIPTION
**Pull request by polybuildr**
Resolves #5.

### CODE REVIEW

**Comment by polybuildr** [comment]
Please test before merging. Don't have a setup local environment.

File: gulpfile.js [hunk 1]
```diff
@@ -52,4 +52,12 @@ gulp.task('stuff', function() {
         .pipe(gulp.dest('build/'));
 });
 
+gulp.task('watch', function() {
+    gulp.watch('src/static/styles/**/*.{scss,css}', ['styles']);
+    gulp.watch('src/static/scripts/**/*.js', ['scripts']);
+    gulp.watch('src/static/images/*.{jpg,jpeg,png,svg}', ['images']);
+    gulp.watch('src/**/*.php', ['watch']);
```
Comment (nisargjhaveri) [inline gulpfile.js]: Is this correct? Wouldn't this be recursive?

Comment (polybuildr) [inline gulpfile.js]: This is a typo. It should be `['php']`.

**Comment by polybuildr** [comment]
Fixed the typo in updated PR.

**Comment by polybuildr** [comment]
Also rebased on master and updated README.md.

**Comment by polybuildr** [comment]
Could someone please test and review this?

**Comment by nisargjhaveri** [comment]
I tried this, but could not test properly as we don't have much content to process yet, and I could not produce that much. But looked like it works.

I would suggest to merge it, and improve it as we use it.

**Comment by polybuildr** [comment]
Fair point. There are a couple of possible enhancements I think will be necessary once actual work starts. Only work that remains is whether we merge now or when it's actually necessary.

I'm okay with now, because it won't break anything. Yeah, we can improve as and when required.

If you're okay with that too, then we can merge.

**Comment by nisargjhaveri** [comment]
Wait.. Does this remove files which are removed in `src/`? If not, can you please look at that also before merging?

**Comment by polybuildr** [comment]
It does not. :P That was on that list I was talking about. Even new files aren't automatically noticed by it. -_-

Alright then, we'll leave this pending until those features can be implemented. :)

**Comment by polybuildr** [comment]
I managed to successfully implement removing of deleted files (that sounds so weird) by using a [gulp recipe](http://fettblog.eu/gulp-recipes-part-2/) I found. ([gist](https://gist.github.com/polybuildr/c4a96fb03bab17e2e642)) However, other issues with `gulp.watch` keep biting, such as its weird ability to miss new files sometimes.

As such, I'm going to take a look at `gulp-watch` to see if that fares any better. Hopefully it'll be more stable (uses `chokidar` instead of an under-renovation `gaze`).

**Comment by polybuildr** [comment]
So apparently watching is so hard no one can do it correctly. `chokidar` has issues too (paulmillr/chokidar#292) and whatever we pick, we're going to come across an issue anyway. I think the previous gist I linked to handles modified files and deleted files just fine. I'll update my PR, you can test it a bit, and then we can probably merge this.

If someone has a more satble solution, please let me know and we can use that instead. The issue in the current gist implementation basically is:
- Don't have an `.htaccess`, `humans.txt` or `robots.txt`.
- Run `rm -rf build/ && gulp && gulp watch`
- Add one of those files
- Nothing happens in `build/`

**Comment by polybuildr** [comment]
Two updates:
- There's an issue in the previous pull request involving files whose extensions also change from `src/` to `build/`.
- I might have done something wrong, but `gulp-watch` is doing even worse than `gulp.watch` with respect to new files.

**Comment by polybuildr** [comment]
- Fixed deletion of files with `.scss` extension.

**Comment by polybuildr** [comment]
Should be okay to merge after a local test by someone who's not me.

**Comment by polybuildr** [comment]
Updated PR:
- Rebased on master.

**Comment by hharchani** [comment]
When running gulp watch, If I create a new scss file, a blank scss file woth same name is created in build folder.

**Comment by hharchani** [comment]
Also, what if I have a.css and a.scss in src/static/styles

**Comment by hharchani** [comment]
If I change one scss file, are all the files currently processed and copied to build?

**Comment by hharchani** [comment]
http://pastebin.com/H17sX3gQ

**Comment by polybuildr** [comment]
@hharchani Thanks for the review!
- The blank file issue has been filed as #14. It is certainly possible to fix it, maybe using something like [gulp-clip-empty-files](https://www.npmjs.com/package/gulp-clip-empty-files). However, I'm not sure it's much of an issue. A stray, empty `.scss` file can't really cause any harm, can it?
- Currently, having `file.css` and `file.scss` is dangerous. Per my tests, AFAICS, since the glob we're using is `{sccs,css}`, the CSS file will be the one that gets copied to `build/` (the opposite happens if the glob is changed). This seems to be pretty weird behaviour, I would have thought the order depended on while file came later in the OS's filesystem. Either way, we know for sure that one of them will overwrite the other. This might be fixable, but I think it would be pretty complex. If you think it should be addressed, please file another issue and we can discuss the specific case there. Personally, I don't think this will be too much of an issue.
- There is currently no layer that checks which files have been updated. I have filed #17 for this.

**Comment by polybuildr** [comment]
Ping?

**Comment by nisargjhaveri** [comment]
There are some merge conflicts. Please have a look? :P

**Comment by polybuildr** [comment]
Sorry, merging of #22 led to conflicts. Fixing.

**Comment by hharchani** [comment]
Haven't accounted for .gif though in images, but still I merged :stuck_out_tongue:

**Pull request merged**